### PR TITLE
Flannel: change default backend type

### PR DESCRIPTION
**Pull request by justinsb**
We support udp, which has to the default for backwards-compatibility,
but also new clusters will now use vxlan.

### CODE REVIEW

File: upup/pkg/fi/cloudup/template_functions.go [hunk 1]
```diff
@@ -105,6 +106,15 @@ func (tf *TemplateFunctions) AddTo(dest template.FuncMap) {
 	}
 
 	dest["ProxyEnv"] = tf.ProxyEnv
+
+	if tf.cluster.Spec.Networking != nil && tf.cluster.Spec.Networking.Flannel != nil {
+		flannelBackendType := tf.cluster.Spec.Networking.Flannel.Backend
+		if flannelBackendType == "" {
+			glog.Warningf("Defaulting flannel backend to udp (not a recommended configuration)")
```
Comment (chrislovecnm) [inline upup/pkg/fi/cloudup/template_functions.go]: Should we tell the user what to do?  You need to set this, or you need to migrate to this?  We are warning them, but not helping them on what to do.

File: pkg/apis/kops/networking.go [hunk 2]
```diff
@@ -60,6 +60,8 @@ type WeaveNetworkingSpec struct {
 
 // Flannel declares that we want Flannel networking
 type FlannelNetworkingSpec struct {
+	// Backend is the backend overlay type we want to use
```
Comment (chrislovecnm) [inline pkg/apis/kops/networking.go]: What are the possible backend values?  Should we have them in the comments?

Comment (justinsb) [inline pkg/apis/kops/networking.go]: Added values to comment

File: upup/pkg/fi/cloudup/template_functions.go (hunk 1 above)
Comment (justinsb) [inline upup/pkg/fi/cloudup/template_functions.go]: Added a release note.  The docs we have for networking are not terribly instructive at the moment... that might have to be another PR though.

**Comment by justinsb** [comment]
Comments addressed - PTAL

**Comment by justinsb** [comment]
Passing tests - PTAL

**Comment by justinsb** [comment]
Reapplying LGTM after rebase

File: upup/pkg/fi/cloudup/template_functions.go (hunk 1 above)
Comment (mstuparu) [reply upup/pkg/fi/cloudup/template_functions.go]: Upgrade path I followed today on a running cluster. Be aware that this creates downtime/connectivity issues ~1 minute. Steps followed:

1. `kops edit cluster` <- change udp for vxlan
2. `kubectl edit configmap -n kube-system kube-flannel-cfg` <- change udp for vxlan
3. `kubectl delete pod -n kube-system -l app=flannel`
